### PR TITLE
[STARTAI-15] Upgrade linear regression

### PR DESCRIPTION
**Pull request by nasstis**

### CODE REVIEW

File: linear/linear_regression_multiple_features.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
+import numpy as np
+
+
+class LinearRegressionMultipleFeatures:
```
Comment (ArsenOOO7) [inline linear/linear_regression_multiple_features.py:4]: Format

**Comment by ArsenOOO7** [comment]
Pls check if matplotlib can build such graphs
If it's possible, add new static function to our visualize helper
And draw it after
зображення

**Review by mesnuk — APPROVED** [review]

File: linear/linear_regression_multiple_features.py [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    
+    def __init__(self, n_estimators, learning_rate):
+        self.__n_estimators = n_estimators
+        self.__learning_rate = learning_rate
+        self.m = None
+        self.b = None
+        self.loss_history = []
```
Comment (JJerome-NM) [inline linear/linear_regression_multiple_features.py:11]: I think `m, b and loss_history` can be private?
Also, are you sure that they should be declared in the init?

File: linear/linear_regression_multiple_features.py [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+    def gradient_descent(self, X, y):
+        predictions = self.predict(X)
+        errors = predictions - y.reshape(-1)
+
+        self.m -= self.__learning_rate * (1/X.shape[0]) * X.T.dot(errors)
+        self.b -= self.__learning_rate * (1/X.shape[0]) * np.sum(errors)
```
Comment (JJerome-NM) [inline linear/linear_regression_multiple_features.py:18]: I think the formatting is still not exactly right

**Review by JJerome-NM — CHANGES_REQUESTED** [review]
In general, everything looks good, but think about the proposed changes

**Review by JJerome-NM — APPROVED** [review]
Look good

**Pull request merged**